File: insight-brain-java-functional-test/src/main/java/com/sonatype/clm/testing/functional/elements/IqBackButton.java
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */

package com.sonatype.clm.testing.functional.elements;

import com.sonatype.clm.testing.functional.BasicElement;

public class IqBackButton
    extends BasicElement<IqBackButton>
{
  /**
   * @param context the context selector, within which the back-button is located
   */
  public IqBackButton(String context) {
    super(context, ".iq-back-button");
  }

  @Override
  public IqBackButton click() {
    child("a").click();
    return me();
  }
}

File: insight-brain-frontend/src/main/frontend/components/iqBackButton/iqBackButton.js
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
import template from './iqBackButton.html';

var iqBackButtonComponent = {
  template,
  controller,
  controllerAs: 'vm',
  bindings: {
    stateName: '@state',
    text: '@?',
  },
};

export default iqBackButtonComponent;

function controller($state) {
  var vm = this;
  vm.linkText = undefined;

  var stateObj = $state.get(this.stateName);
  if (stateObj) {
    if (this.text !== undefined) {
      vm.linkText = this.text;
    } else if (stateObj.data && stateObj.data.title) {
      vm.linkText = 'Back to ' + stateObj.data.title;
    } else {
      vm.linkText = 'Back';
    }
  } else {
    throw new Error(
      'Failed to display iq-back-button, provided state does not exist: ' +
        this.stateName
    );
  }
}

controller.$inject = ['$state'];
